portfolio/src/main/webapp/js/script.js

```diff
@@ -38,21 +38,41 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
+async function initializeComments() {
+  await getComments();
+  const response = await fetch('/auth-status');
+  const authData = await response.json();
+
+  if (!authData.isLoggedIn) {
+    const commentCreateForm = document.getElementById('comment-form');
+    commentCreateForm.hidden = true;
+    const logoutContainer = document.getElementById('logout-container');
+    logoutContainer.hidden = true;
+    const loginUrl = document.getElementById('login-url');
+    loginUrl.innerHTML = `<a href="${authData.authURL}"> here </a>`;
```

Comment:
innerHTML can cause security issues. Make sure you're taking care of them. 
https://gomakethings.com/preventing-cross-site-scripting-attacks-when-using-innerhtml-in-vanilla-javascript/